buffer/src/main/java/io/netty/buffer/AbstractPooledDerivedByteBuf.java

```diff
@@ -43,6 +43,12 @@
         this.recyclerHandle = (Handle<AbstractPooledDerivedByteBuf>) recyclerHandle;
     }
 
+    // Called from within SimpleLeakAwareByteBuf and AdvancedLeakAwareByteBuf.
+    final void parent(ByteBuf newParent) {
+        assert newParent instanceof SimpleLeakAwareByteBuf || newParent instanceof AdvancedLeakAwareByteBuf;
```

Comment:
@Scottmitch I could even change this to `assert newParent instanceof SimpleLeakAwareByteBuf` as `AdvancedLeakAwareByteBuf`now extends `SimpleLeakAwareByteBuf `. Not sure if its better to be explicit here tho. WDYT ?